staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -227,18 +231,41 @@ func (s *Controller) Run(stopCh <-chan struct{}, workers int) {
 		go wait.Until(s.worker, time.Second, stopCh)
 	}
 
-	go wait.Until(s.nodeSyncLoop, nodeSyncPeriod, stopCh)
+	go wait.Until(func() { s.nodeSyncLoop(workers) }, time.Second, stopCh)
```

Comment:
you don't need wait.Until here - the nodeSyncLoop() runs until nodeSyncCh is open anyway and will never close.
So stopChn will not be effective to it anyway too.

so you can simply change to:

`go nodeSyncLook(workers)`

Reply:
done